Automattic/vip-go-mu-plugins
# Use sampling to reduce Search stat collection

## Description

Presently we're sending too many stats too quickly. We'll start with a simple sampling and adjust as needed.

The main Search class and the Queue class got their own basic implementation of sampling so the tests could be adjusted to work properly without rewiring everything.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

1. Perform 10 search actions.
2. Every search action should record stats.
3. Apply PR.
4. Perform 10 search actions.
5. Only roughly half of the actions should record stats.
